tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/CDIBean.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrency.tck.cdi;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.*;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+import org.eclipse.microprofile.concurrent.NamedInstance;
+import org.testng.Assert;
+
+@RequestScoped
+public class CDIBean {
+
+    @Inject
+    ManagedExecutor exec;
+
+    @Inject
+    @NamedInstance("maxAsync2")
+    @ManagedExecutorConfig(maxAsync = 2)
+    ManagedExecutor maxAsync2;
+
+    @Inject
+    @NamedInstance("appProduced")
+    ManagedExecutor appProduced;
+
+    @Inject
+    @NamedInstance("appProduced_injected")
+    ManagedExecutor appProduced_injected;
+
+    @Inject
+    ManagedExecutor throwAway1;
+
+    @Inject
+    ManagedExecutor throwAway2;
+
+    @Produces
+    @ApplicationScoped
+    @NamedInstance("appProduced")
+    public ManagedExecutor createExec() {
+        return ManagedExecutor.builder().build();
+    }
+
+    @Produces
+    @ApplicationScoped
+    @NamedInstance("appProduced_injected")
+    public ManagedExecutor createExecInjected(@ManagedExecutorConfig(maxAsync = 2) ManagedExecutor exec) {
+        System.out.println("@AGG in producer with exec: " + exec);
+        Thread.dumpStack();
+        return exec;
+    }
+
+    /**
+     * Verify that injected ME instances injected by the container can be shutdown
+     */
+    public void shutdownContainerInstance() throws Exception {
+        throwAway1.shutdown();
+        assertTrue(throwAway1.awaitTermination(5, TimeUnit.SECONDS));
```

Comment:
Recommend using the same default maximum allowable timeout as other test cases in the MP Concurrency bucket.  Also applies a few lines later